Cesium.CodeGen/Ir/Expressions/BinaryOperators/ArithmeticBinaryOperatorExpression.cs

```diff
@@ -71,6 +51,64 @@ public override IExpression Lower(IDeclarationScope scope)
         return new ArithmeticBinaryOperatorExpression(left, Operator, right);
     }
 
+    private IExpression LowerPointerArithmetics(IDeclarationScope scope, IExpression left, IExpression right, IType leftType, IType rightType)
+    {
+        if (leftType is PointerType leftPointerType)
+        {
+            if (rightType is PointerType rightPointerType)
+            {
+                if (Operator != BinaryOperator.Subtract)
+                {
+                    throw new CompilationException($"Operator {Operator} is not supported for pointer/pointer operands");
+                }
+
+                var baseSizeLeft = leftPointerType.Base.GetSizeInBytes(scope.ArchitectureSet);
+                var baseSizeRight = rightPointerType.Base.GetSizeInBytes(scope.ArchitectureSet);
+
+                if (baseSizeLeft == null || baseSizeRight == null || baseSizeLeft != baseSizeRight)
```

Comment:
I suggest directly comparing the types instead of comparing their sizes: the current implementation will allow subtracting pointers to different types of identical size, which shouldn't be possible.

Reply:
Done